src/features/docSlices.ts

```diff
@@ -19,17 +19,61 @@ export type Point = {
   x: number;
 };
 
-export type Shape = {
+export interface Box {
+  y: number;
+  x: number;
+  width: number;
+  height: number;
+}
+
+// TODO(ppeeou) refer to yorkie-sdk-js ArrayProxy
+export interface BaseShape {
   type: string;
+  getID(): TimeTicket;
+}
+
+export interface Line extends BaseShape {
+  type: 'line';
+  color: string;
   points: Array<Point>;
-};
+}
+
+export interface EraserLine extends BaseShape {
+  type: 'eraser';
+  points: Array<Point>;
+}
+
+export interface Rect extends BaseShape {
+  type: 'rect';
+  points: Array<Point>;
+  box: Box;
+}
+
+export type Shape = Line | EraserLine | Rect;
 
 export type PairDoc = {
   mode: CodeMode;
   content: PlainText;
   shapes: Array<Shape>;
 };
 
+// TODO(ppeeou) refer to yorkie-sdk-js ArrayProxy
+export interface Shapes {
+  push(shape: Shape): number;
+
+  getLast(): Shape;
+  getElementByID(createdAt: TimeTicket): Shape;
+  deleteByID(createdAt: TimeTicket): Shape;
+  length: number;
+  [index: number]: Shape;
+  [Symbol.iterator](): IterableIterator<Shape>;
+}
+
+//  TODO(ppeeou) refer to yorkie-sdk-js JSONObject
```

Comment:
It would be nice to express what kind of work you are going to do in the future.

Reply:
It just means I'll use the type from `yorkie-sdk-js`, nothing else..!

Reply:
It is better to register this issue as a separate issue.

Reply:
Create related issue https://github.com/yorkie-team/yorkie-codepair/issues/110